support-lib/cpp/expected.hpp

```diff
@@ -1,15 +1,27 @@
 #pragma once
 
+#include <version>
```

Comment:
`<version>` is a c++20 header.  Can we somehow maintain compatible with c++17?

Reply:
do you think we can guard it with `__has_include`?

Reply:
Hi,

Please, use:

```cpp
#if __cplusplus >= 202002L
#include <version>
#endif
```

It will make compatible with c++17

Thanks.

Reply:
Thank you for the feedback, this makes sense. I've applied the change suggested by @paulocoutinhox and also added a `#ifdef __has_include` before it, because this should make it compatible with even earlier compilers. I got the idea from the example in https://en.cppreference.com/w/cpp/feature_test , which is apparently even C++11 compatible.

Reply:
The code from cppreference is:

```cpp
#ifdef __has_include
# if __has_include(<version>)
#   include <version>
# endif
#endif
```

Can be this too.

Reply:
You mean the indentations? I don't usually use that style, so I removed it without thinking about it.

I'm trying to figure out what's usually used in this codebase. Looking at `DataRef.hpp` and `Future.hpp`, there are sometimes indentations for precompiler directives, but not always. And the indentations are sometimes 2, sometimes 4 spaces. And the indentations are always before the pound sign, never after. I'll happily change the indentations if someone (@LiFengSC ?) can tell me what the preferred style is.

Reply:
No, i mean the two directives for secure include it. 

First is checking if exists __has_include: `#ifdef __has_include`
Second is checking the version header: `#if __has_include(<version>)`